classes/Collection.js

```diff
@@ -146,28 +146,20 @@ class Collection {
       const newNavContent = base64.encode(yaml.safeDump(newNavContentObject))
       await nav.update(NAV_FILE_NAME, newNavContent, navSha)
 
-      // Get all collectionPages
-      const OldIsomerFile = new File(this.accessToken, this.siteName)
-      const oldCollectionPageType = new CollectionPageType(oldCollectionName)
-      OldIsomerFile.setFileType(oldCollectionPageType)
-      const collectionPages = await OldIsomerFile.list()
-      
-      // If the object is empty (there are no pages in the collection), do nothing
-      if (_.isEmpty(collectionPages)) return 
-
-      // Set up new collection File instance
-      const NewIsomerFile = new File(this.accessToken, this.siteName)
-      const newCollectionPageType = new CollectionPageType(newCollectionName)
-      NewIsomerFile.setFileType(newCollectionPageType)
-
-      // Rename all collectionPages
-      await Bluebird.map(collectionPages, async(collectionPage) => {
-        let pageName = collectionPage.fileName
-        const { content, sha } = await OldIsomerFile.read(pageName)
-        await OldIsomerFile.delete(pageName, sha)
-        return NewIsomerFile.create(pageName, content)
+      const { gitTree, currentCommitSha } = await getRootTree(this.siteName, this.accessToken);
```

Comment:
One thing to note is that we have an atomicity problem in this endpoint: it's possible for one of the API calls to succeed and the rest to fail (for example, config is updated, but nav is not), leading to an inconsistent state. 

I think one way we can fix this is that we retrieve the git tree first so that we have the current commit sha that our branch is pointing to. And then whenever there is an error, we can point the branch back to the original commit sha with an API call in the catch block.

Reply:
Ideally, it would have been great if we could update the pages all together with the git tree, but git tree does not allow you to modify the blob contents of individual files in a single api call

Reply:
Let's do this in a separate PR for clarity

Reply:
As discussed, we will be resolving this in a separate PR